synfig-studio/src/gui/widgets/widget_timetrack.cpp

```diff
@@ -234,6 +234,16 @@ bool Widget_Timetrack::copy_selected(synfig::Time delta_time)
 	return ok;
 }
 
+void Widget_Timetrack::handle_copied_waypoints_selection()
+{
+	if (!waypoint_sd.is_selected(waypoint_sd.get_hovered_item())){ //if copied waypoint isnt selected select it and deselect any other waypoints
+		std::vector<WaypointItem*> selection = waypoint_sd.get_selected_items();
+		for(WaypointItem *wi : selection)
+			waypoint_sd.deselect(*wi);
```

Comment:
Anyway, I think it would be good to have a "waypoint clipboard", in order to allow multiple pastes. Or does it already allow?

Reply:
I believe multiple pastes is already available, this was so that if the copied waypoint isn't selected then it gets selected so that it's copied, and other selected waypoints are deselected. Unless the copied waypoint is part of the multiple selected waypoints then in this case the user wants to copy/paste multiple waypoints.

Reply:
> maybe create a method `SelectDragHelper::clear_selection()`? Reduces the number of emitted signals.

makes sense, no point in emitting many selection-changed signals when we know we want to deselect all.